ddtrace/utils/formats.py

```diff
@@ -97,7 +98,25 @@ def parse_tags_str(tags_str):
     if not tags_str:
         return parsed_tags
 
-    for tag in tags_str.split(","):
+    # Identify separator based on which successfully identifies the correct
+    # number of valid tags
+    numtagseps = tags_str.count(":")
+    for sep in [",", " "]:
+        if sum(1 for _ in tags_str.split(sep) if _.count(":") == 1) == numtagseps:
+            break
+    else:
+        log.error(
```

Comment:
From looking at internal usage, I am still team error:

```
$ git grep parse_tags_str
ddtrace/bootstrap/sitecustomize.py:25:from ddtrace.utils.formats import parse_tags_str
ddtrace/bootstrap/sitecustomize.py:63:    modules = parse_tags_str(modules_to_patch)
ddtrace/bootstrap/sitecustomize.py:113:        tracer.set_tags(parse_tags_str(env_tags))
ddtrace/profiling/exporter/http.py:18:from ddtrace.utils.formats import parse_tags_str
ddtrace/profiling/exporter/http.py:65:                parse_tags_str(os.environ.get("DD_TAGS")).items(),
ddtrace/profiling/exporter/http.py:66:                parse_tags_str(os.environ.get("DD_PROFILING_TAGS")).items(),
ddtrace/settings/config.py:9:from ..utils.formats import parse_tags_str
ddtrace/settings/config.py:56:        self.tags = parse_tags_str(os.getenv("DD_TAGS") or "")
ddtrace/settings/config.py:65:        self.service_mapping = parse_tags_str(get_env("service", "mapping", default=""))
```

But we can handle as a follow-up.